src/cli/sql.rs

```diff
@@ -246,6 +246,10 @@ pub async fn init(
 			}
 		}
 	}
+	// drop the printer channel so the printer task will quit.
+	std::mem::drop(tx);
+	// print task shouldn't panic or be cancelled so this unwrap should never trigger.
+	print_task.await.unwrap();
```

Comment:
I think panicking is better for catching things like these. Log messages can easily be overlooked or they may not be enabled at all. `unwrap()`, however, doesn't communicate that "this shouldn't happen". The comment will help of course but I prefer `unreachable` for this because it communicates intention clearly.

```rust
print_task.await.unwrap_or_else(|_| unreachable!("print task shouldn't panic or be cancelled"));
```

With unreachable you know right away that we reached a part of the code that shouldn't be possible to reach even without including that message in the macro.

Reply:
I disagree unwrap does communicate that `this shouldn't happen` in fact you should only call unwrap on things that shouldn't happen. The comment is there to clarify why it shouldn't happen.

Reply:
I like unwrapping in these cases because the unwrap indicates an assumption, a valid assumption in this case, but this might change in the future if other code changes. A panic happening will then clearly indicate that some kind of bug has been created.

Reply:
I know what you are saying @DelSkayn and I agree that one should never use `unwrap` in production code unless the panic effectively indicates a bug. Some people use it for prototyping though, the idea being to be able to quickly implement and test an idea without having to worry about error handling. They would then go back and clean up those unwraps by handling the error properly once the idea is validated.

My point was, when you encounter an unwrap you can't tell right away whether it happened because someone forgot to handle an error properly or not. Yes, when they dive into the code they will know that when they read your comment. If you forgot to put that comment in or if someone subsequently removes it, the mindset needed for debugging this would be "undefined". Some people might simply decide to handle it as an error, while some may see it as a bug that it is.

`unreachable!`, even without any comments or panic messages, communicates very clearly that a bug happened. You know right away even before jumping into the code. This is why I like using `unreachable` in production code for pointing out bugs like this. I use `unwrap` in tests because it's more convenient to write and there is no potential for it being interpreted incorrectly.

Reply:
Just to be clear, I think `unreachable!` would be a slight improvement but I'm fine with merging this as it is too. There is no need to block this PR on that.